core/src/test/scala/integration/kafka/api/ConsumerBounceTest.scala

```diff
@@ -147,6 +148,230 @@ class ConsumerBounceTest extends IntegrationTestHarness with Logging {
+  }
+
+  /**
+   * Consumer using manual partition assignment is closed while coordinator is unavailable,
+   * but other brokers are available. Close should complete gracefully with successful commits
+   * for manual partition assignment.
+   */
+  @Test
+  def testCloseWithCoordinatorFailureUsingManualAssignment() {
+    val numRecords = 10
+    sendRecords(numRecords)
+    val consumer = this.consumers.remove(0)
+    consumer.assign(Collections.singletonList(tp))
+    receiveRecords(consumer, numRecords)
+    val adminClient = AdminClient.createSimplePlaintext(this.brokerList)
+    val coordinator = adminClient.findCoordinator("my-test").id
+    killBroker(coordinator)
+    closeAndValidate(consumer, Long.MaxValue, None, Some(1000))
+    checkClosedState(consumers.head, numRecords)
+  }
+
+  /**
+   * Consumer is closed with a small timeout while all brokers are unavailable.
+   * Cannot rebalance or commit offsets since there is no coordinator, but close should
+   * timeout and return.
+   */
+  @Test
+  def testCloseTimeoutWithClusterFailure() {
+    val numRecords = 10
+    sendRecords(numRecords)
+    val consumer = this.consumers.remove(0)
+    consumer.subscribe(Collections.singletonList(topic))
+    receiveRecords(consumer, numRecords)
+    servers.foreach(server => killBroker(server.config.brokerId))
+    val closeTimeout = 2000
+    closeAndValidate(consumer, closeTimeout, Some(closeTimeout), Some(closeTimeout))
+  }
+
+  /**
+   * Consumer is closed without a timeout while all brokers are unavailable.
+   * Cannot commit offsets since there are no brokers, but close should
+   * terminate on request timeout after attempting to commit offsets even
+   * though offsets cannot be successfully committed.
+   */
+  @Test
+  def testCloseNoTimeoutWithClusterFailure() {
+    val numRecords = 10
+    sendRecords(numRecords)
+    val requestTimeout = 6000
+    this.consumerConfig.setProperty(ConsumerConfig.SESSION_TIMEOUT_MS_CONFIG, "5000")
+    this.consumerConfig.setProperty(ConsumerConfig.HEARTBEAT_INTERVAL_MS_CONFIG, "1000")
+    this.consumerConfig.setProperty(ConsumerConfig.REQUEST_TIMEOUT_MS_CONFIG, requestTimeout.toString)
+    val consumer = createNewConsumer
+    consumer.assign(Collections.singletonList(tp))
+    receiveRecords(consumer, numRecords)
+    servers.foreach(server => killBroker(server.config.brokerId))
+    closeAndValidate(consumer, Long.MaxValue, Some(requestTimeout), Some(requestTimeout))
+  }
+
+  /**
+   * Consumer is closed during rebalance. Close should leave group and close
+   * immediately if rebalance is in progress.
+   */
+  @Test
+  def testCloseDuringRebalance() {
+    val executor = Executors.newSingleThreadExecutor
+    try {
+      closeDuringRebalance(executor, true)
+    } finally {
+      executor.shutdownNow()
+    }
+  }
+
+  /**
+   * Consumer closed during rebalance when brokers are not available.
+   * Close should terminate immediately since coordinator not known.
```

Comment:
The two tests uses a close timeout of Long.MaxValue. The timeout tested is 1 second (to process leave group in the case where brokers are available) + a grace period - if close doesn't complete within this timeout, the test fails.  I have updated comments in the test and also changed to using a variable instead of 1000ms. Hopefully it makes it clearer now.